website/client/src/pages/private-messages.vue

```diff
@@ -630,7 +630,16 @@ export default {
 
     await this.reload();
 
-    const data = this.$store.state.privateMessageOptions;
+    let data = this.$store.state.privateMessageOptions;
```

Comment:
I'd say this is the perfect place to use Navigation Guards https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards, they'll be called when the route is loaded